Make fails when NDEBUG=1

## Steps to reproduce the issue
make clean
make config-clang
# set NDEBUG=1 in Makefile with your preferred editor
make

## Expected behavior

Build success

## Actual behavior

Build fails with:

```
passes/techmap/abc9_ops.cc:778:21: error: use of undeclared identifier 'box_module'
                        auto abc9_flop = box_module->attributes.count("\\abc9_flop");
                                         ^
passes/techmap/abc9_ops.cc:789:46: error: use of undeclared identifier 'derived_type'
                        for (const auto &port_name : box_ports.at(derived_type)) {
                                                                  ^
passes/techmap/abc9_ops.cc:790:22: error: use of undeclared identifier 'box_module'
                                RTLIL::Wire *w = box_module->wire(port_name);
```

--- a/passes/techmap/abc9_ops.cc
+++ b/passes/techmap/abc9_ops.cc
@@ -752,13 +752,11 @@ void reintegrate(RTLIL::Module *module)
 				continue;
 			}
 
-#ifndef NDEBUG
 			RTLIL::Module* box_module = design->module(existing_cell->type);
 			IdString derived_type = box_module->derive(design, existing_cell->parameters);
 			RTLIL::Module* derived_module = design->module(derived_type);
 			log_assert(derived_module);
 			log_assert(mapped_cell->type == stringf("$__boxid%d", derived_module->attributes.at("\\abc9_box_id").as_int()));
-#endif
 			mapped_cell->type = existing_cell->type;
 
 			RTLIL::Cell *cell = module->addCell(remap_name(mapped_cell->name), mapped_cell->type);

